rustls/src/vecbuf.rs

```diff
@@ -36,6 +36,10 @@ impl ChunkVecBuffer {
         self.chunks.is_empty()
     }
 
+    pub(crate) fn is_full(&self) -> bool {
+        self.len() > Self::MAX_SIZE
```

Comment:
suggest we make this ```
if let Some(limit) = self.limit {
  limit <= self.len()
} else {
  False
}```
And then construct `received_plaintext` with `Some(16384)` in conn.rs. That keeps all the mechanism in this file, and all the policy in conn.rs.

Reply:
Sounds good, will take care of it!